sql/connect/server/src/main/scala/org/apache/spark/sql/connect/ml/MLHandler.scala

```diff
@@ -112,12 +113,37 @@ private object ModelAttributeHelper {
 
 // MLHandler is a utility to group all ML operations
 private[connect] object MLHandler extends Logging {
+
+  val currentSessionHolder = new java.lang.ThreadLocal[SessionHolder] {
+    override def initialValue: SessionHolder = null
+  }
+
+  val safeMLClassLoader: String => Class[_] = {
+    val transformerClasses = MLUtils.loadOperators(classOf[Transformer])
+    val estimatorClasses = MLUtils.loadOperators(classOf[Estimator[_]])
+    val evaluatorClasses = MLUtils.loadOperators(classOf[Evaluator])
+    val whitelistedClasses = (
+      transformerClasses ++ estimatorClasses ++ evaluatorClasses
+        ++ Map(
+        "org.apache.spark.ml.clustering.PowerIterationClustering" ->
+          classOf[org.apache.spark.ml.clustering.PowerIterationClustering])
+      )
+
+    (className: String) => {
+      val sessionHolder = currentSessionHolder.get()
+      assert(sessionHolder != null)
```

Comment:
`safeMLClassLoader` is a val so it will be eagerly initialized. How do we make sure `def handleMlCommand` is invoked earlier to set the session holder?